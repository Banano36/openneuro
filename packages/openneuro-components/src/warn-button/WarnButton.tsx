import React, { useRef, useEffect } from 'react'
import './warn-button.scss'
import { Tooltip } from '../tooltip/Tooltip'
import { Button } from '../button/Button'

export interface WarnButtonProps {
  message?: string
  icon?: string
  disabled?: boolean
  tooltip?: string
  onConfirmedClick?: () => void
  displayOptions: boolean
  setDisplayOptions(cb: (currentState: boolean) => boolean): void
}
export const WarnButton = ({
  message,
  icon,
  disabled,
  tooltip,
  onConfirmedClick,
  displayOptions,
  setDisplayOptions,
}: WarnButtonProps) => {
  const viewAction = (
    <div className="warn-btn-group " role="group">
      <div className="slide-in">
        <Button
          className="btn-warn-component cancel"
          iconSize="12px"
          icon="fa fa-times"
          color="#fff"
          backgroundColor="#c00342"
          onClick={() => setDisplayOptions(currentState => !currentState)}
          size="xsmall"
        />

        <Button
          className="btn-warn-component success"
          iconSize="12px"
          icon="fa fa-check"
          color="#fff"
          backgroundColor="#00b489"
          onClick={() => {
            onConfirmedClick()
            setDisplayOptions(currentState => !currentState)
          }}
          size="xsmall"
        />
      </div>
    </div>
  )

  const hideAction = (
    <span className={disabled ? ' disabled warn-btn-click' : 'warn-btn-click'}>
      <Button
        className="btn-warn-component"
        iconSize="12px"
        icon={'fa ' + icon}
        label={message}
        onClick={() => setDisplayOptions(currentState => !currentState)}
        disabled={disabled}
      />
    </span>
  )

  const button = displayOptions ? viewAction : hideAction

  return (
    <div className="warn-btn">
      {tooltip ? (
        <Tooltip flow="up" tooltip={tooltip}>
          {button}
        </Tooltip>
      ) : (
        button
      )}
    </div>
  )
}
