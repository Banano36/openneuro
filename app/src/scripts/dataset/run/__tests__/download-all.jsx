import React from 'react'
import { shallow } from 'enzyme'
import DownloadAll from '../download-all.jsx'
import WarnButton from '../../../common/forms/warn-button.jsx'
import DownloadS3 from '../download-s3.jsx'

const dsId = '59c16458c650bd001111f010'
const sId = '3030313030332d3030303031'

describe('dataset/run/DownloadAll', () => {
  it('should render something', () => {
    expect(
      shallow(<DownloadAll datasetId={dsId} snapshotId={sId} />),
    ).toMatchSnapshot()
  })
  it('should render a WarnButton for downloading via the web', () => {
    expect(
      shallow(<DownloadAll datasetId={dsId} snapshotId={sId} />).find(
        WarnButton,
      ).length,
    ).toBe(1)
  })
  it('should render a DownloadS3 component for downloading via AWS cli', () => {
    expect(
      shallow(<DownloadAll datasetId={dsId} snapshotId={sId} />).find(
        DownloadS3,
      ).length,
    ).toBe(1)
  })
})
