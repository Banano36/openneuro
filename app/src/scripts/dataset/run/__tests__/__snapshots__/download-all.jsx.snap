// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/run/DownloadAll should render something 1`] = `
<span
  className="download-all"
>
  <WarnButton
    cancel={
      <i
        className="fa fa-times"
      />
    }
    confirm={
      <i
        className="fa fa-check"
      />
    }
    icon="fa-download"
    message=" DOWNLOAD All"
    prepDownload={[Function]}
    tooltip={null}
    warn={true}
  />
</span>
`;
